contributing/core/developing-using-local-app.md

```diff
@@ -0,0 +1,70 @@
+# Developing Using Your Local Version of Next.js
+
+There are two options to develop with your local version of the codebase:
+
+### Set as a local dependency in package.json
+
+1. In your app's `package.json`, replace:
+
+   ```json
+   "next": "<next-version>",
+   ```
+
+   with:
+
+   ```json
+   "next": "file:/path/to/next.js/packages/next",
+   ```
+
+2. In your app's root directory, make sure to remove `next` from `node_modules` with:
+
+   ```sh
+   rm -rf ./node_modules/next
+   ```
+
+3. In your app's root directory, run:
+
+   ```sh
+   pnpm i
+   ```
```

Comment:
IMHO this could be re-worked.

Usually, when I try to work on an existing bug, I'd clone their reproduction and link my local next in them. The reproduction is most likely not using pnpm. So instead of simply `pnpm i`, I almost always have to do:

```sh
pnpm import # This migrating existing yarn.lock / package-lock.json to pnpm-lock.yaml
rm -rf node_modules .yarn .yarnrc yarn.lock package-lock.json
pnpm i
```

...in order to debug it.

Reply:
Continuing the discussion here: https://github.com/vercel/next.js/pull/40515#discussion_r969623609